samcli/lib/sync/flows/layer_sync_flow.py

```diff
@@ -202,6 +201,16 @@ def __init__(
         super().__init__(layer_identifier, build_context, deploy_context, sync_context, physical_id_mapping, stacks)
         self._layer = cast(LayerVersion, build_context.layer_provider.get(self._layer_identifier))
 
+    @property
+    def sync_state_identifier(self) -> str:
+        """
+        Sync state is the unique identifier for each sync flow
+        In sync state toml file we will store
+        Key as LayerSyncFlow:LayerLogicalId
+        Value as layer ZIP hash
+        """
+        return self.__class__.__name__ + ":" + self._layer_identifier
```

Comment:
This could be done in `AbstractLayerSyncFlow` so that we don't have to repeat it for other instances (since they are the same).